File: models/story.js
const mongoose = require('mongoose');

const StorySchema = new mongoose.Schema({
    title:{
        type: String,
        required: true
    },

    category:{
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Category',
        default: null
    },

    rating:{
        type: Number,
        default: 1.0
    },

    age: {
        type: mongoose.Schema.Types.ObjectId,
        default: null,
        ref: 'Age'
    },

    pdf: {
        type: String,
        required: true
    },

    image:{
        type: String,
        required: true
    },

    interactions:{
        type: [String],
        default: []
    },

    comments:{
        type: [String],
        default: [],
        ref: 'Comment'
    },

    publisher:{
        type: String,
        required: true,
    },

    views: {
        type: Number,
        default: 0
    },

    publish_date:{
        type: String,
        required: true
    }
})


const StoryModel = mongoose.model('Story', StorySchema)

module.exports = StoryModel

File: lib/data/models/story.dart
import 'package:log/data/models/age.dart';
import 'package:log/data/models/category.dart';

class Story{
  final String title;
  Category? category;
  Age? age;
  final String file;
  final String image;
  final double rating;
  final List<String> interactions;

  Story({
    required this.rating,
    required this.title, 
    this.category, 
    this.age, required this.file, required this.image, required this.interactions});

  factory Story.fromJson(Map data){
    return Story(
      title: data['title'],
      category: data['category'] == null ? null :  Category.fromJson(data['category']),
      age: data['age'] == null ? null : Age.fromJson(data['age']),
      file: data['pdf'],
      image: data['image'],
      rating: double.parse(data['rating'].toString()),
      interactions: (data['interactions'] as List<dynamic>).map((e) => e.toString()).toList()
    );
  }
}